fv3core/stencils/remapping_part2.py

```diff
@@ -27,125 +27,176 @@ def init_phis(hs: FloatField, delz: FloatField, phis: FloatField, te_2d: FloatFi
-            mdt,
-            fast_mp_consv,
-            last_step,
-            akap,
-            kmp,
+        self._moist_cv_last_step_stencil = FrozenStencil(
+            moist_cv.moist_pt_last_step,
+            origin=(self.grid.is_, self.grid.js, 0),
+            domain=(self.grid.nic, self.grid.njc, self.grid.npz + 1),
         )
-        if fast_mp_consv:
-            sum_te(te, te0_2d, origin=kmp_origin, domain=kmp_domain)
-    if last_step:
-        moist_cv.compute_last_step(
-            pt, pkz, dtmp, r_vir, qvapor, qliquid, qice, qrain, qsnow, qgraupel, gz
+
+        self._basic_adjust_divide_stencil = FrozenStencil(
+            basic.adjust_divide_stencil,
+            origin=self.grid.compute_origin(),
+            domain=self.grid.domain_shape_compute(),
         )
-    else:
-        basic.adjust_divide_stencil(
-            pkz, pt, origin=grid.compute_origin(), domain=grid.domain_shape_compute()
+
+        self.do_sat_adjust = self.namelist.do_sat_adj
+
+        # TODO : When remapping becomes an object, these stencils can be declared in
+        #        __init__ when (I think) pFull can be passed into __init__
+        # self._saturation_adjustment_stencil = FrozenStencil(SatAdjust3d,...)
+        # self._sum_te_stencil = FrozenStencil(sum_te,...)
+
+        # TODO : When state.pt from fv_dynamics can be passed into __init__, these
+        #        storages can be declared in __init__
+        # phis = utils.make_storage_from_shape(
+        #     pt.shape, grid.compute_origin(), cache_key="remapping_part2_phis"
+        # )
+        # te_2d = utils.make_storage_from_shape(
+        #     pt.shape[0:2], grid.compute_origin(), cache_key="remapping_part2_te_2d"
+        # )
+        # zsum1 = utils.make_storage_from_shape(
+        #     pt.shape[0:2], grid.compute_origin(), cache_key="remapping_part2_zsum1"
+        # )
+
+    def __call__(
+        self,
+        qvapor: FloatField,
+        qliquid: FloatField,
+        qice: FloatField,
+        qrain: FloatField,
+        qsnow: FloatField,
+        qgraupel: FloatField,
+        qcld: FloatField,
+        pt: FloatField,
+        delp: FloatField,
+        delz: FloatField,
+        peln: FloatField,
+        u: FloatField,
+        v: FloatField,
+        w: FloatField,
+        ua: FloatField,
+        cappa: FloatField,
+        q_con: FloatField,
+        gz: FloatField,
+        pkz: FloatField,
+        pk: FloatField,
+        pe: FloatField,
+        hs: FloatFieldIJ,
+        te0_2d: FloatFieldIJ,
+        te: FloatField,
+        cvm: FloatField,
+        pfull: FloatFieldK,
+        ptop: float,
+        akap: float,
+        r_vir: float,
+        last_step: bool,
+        pdt: float,
+        mdt: float,
+        consv: float,
+        do_adiabatic_init: bool,
+    ):
+        saturation_adjustment = utils.cached_stencil_class(SatAdjust3d)(
+            cache_key="satadjust3d"
         )
+
+        sum_te_stencil = gtstencil(func=sum_te)
```

Comment:
this can still be done in init, even if you don't have the origin and domain in runtime, which is possible if you want to do that too.

Reply:
Done!